compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirMemberPropertyChecker.kt

```diff
@@ -0,0 +1,131 @@
+            !property.isAbstract &&
+            (property.getter == null || property.getter is FirDefaultPropertyAccessor)
+        ) {
+            property.source?.let { source ->
+                val modifier = source.getModifierList()?.modifiers?.find { it.token == PRIVATE_KEYWORD }
+                if (modifier != null && modifier.source != null) {
+                    reporter.report(FirErrors.PRIVATE_PROPERTY_IN_INTERFACE.on(modifier.source!!, modifier.token))
+                }
+            }
+        }
+
+        if (property.isAbstract) {
+            if (!containingDeclaration.isAbstract && !containingDeclaration.isSealed && !inEnumClass(containingDeclaration)) {
+                property.source?.let { source ->
+                    val modifier = source.getModifierList()?.modifiers?.find { it.token == ABSTRACT_KEYWORD }
+                    if (modifier != null && modifier.source != null) {
+                        reporter.report(FirErrors.ABSTRACT_PROPERTY_IN_NON_ABSTRACT_CLASS.on(modifier.source!!, modifier.token))
+                        return
+                    }
+                }
+            }
+
+            if (property.delegate != null) {
+                property.delegate!!.source?.let {
+                    if (inInterface(containingDeclaration)) {
+                        reporter.report(FirErrors.DELEGATED_PROPERTY_IN_INTERFACE.on(it, property.delegate!!))
+                    } else {
+                        reporter.report(FirErrors.ABSTRACT_DELEGATED_PROPERTY.on(it, property.delegate!!))
+                    }
+                }
+            }
+
+            checkAccessor(property.getter, property.delegate) { src, symbol ->
+                reporter.report(FirErrors.ABSTRACT_PROPERTY_WITH_GETTER.on(src, symbol))
+            }
+            checkAccessor(property.setter, property.delegate) { src, symbol ->
+                if (symbol.fir.visibility == Visibilities.Private && property.visibility != Visibilities.Private) {
+                    reporter.report(FirErrors.PRIVATE_SETTER_FOR_ABSTRACT_PROPERTY.on(src, symbol))
+                } else {
+                    reporter.report(FirErrors.ABSTRACT_PROPERTY_WITH_SETTER.on(src, symbol))
+                }
+            }
+        }
+
+        checkPropertyInitializer(containingDeclaration, property, reporter)
+
+        if (property.isOpen) {
+            checkAccessor(property.setter, property.delegate) { src, symbol ->
+                if (symbol.fir.visibility == Visibilities.Private && property.visibility != Visibilities.Private) {
+                    reporter.report(FirErrors.PRIVATE_SETTER_FOR_OPEN_PROPERTY.on(src, symbol))
+                }
+            }
+        }
+    }
+
+    private fun checkPropertyInitializer(containingDeclaration: FirRegularClass, property: FirProperty, reporter: DiagnosticReporter) {
+        if (property.initializer != null) {
+            property.initializer!!.source?.let {
```

Comment:
`property.initiailizer?.source?.let` and remove the first `if`

Reply:
Done, Thanks.